src/_pytest/python.py

```diff
@@ -596,6 +596,7 @@ def isinitpath(self, path):
     def collect(self):
         this_path = self.fspath.dirpath()
         pkg_prefix = None
+        yield Module(this_path.join("__init__.py"), self)
```

Comment:
as is this isnt acceptable, since it doesn't check if the filenames is acceptable and might cause surprises for people that have objects with fitting names in packages

Reply:
Ahh good catch @RonnyPfannschmidt. Just to be clear you mean to change this bit to something like:

```python
init_module = this_path.join("__init__.py")
if init_module.check(file=1) and init_module.match(self.config.getini('python_files')):
    yield Module(init_module, self)
```

?

Reply:
we should reuse the filename matching and we dont need to check for existence anymore

Reply:
Right, thanks!

Reply:
Well, the `__init__.py` is hardcoded at:
https://github.com/pytest-dev/pytest/blob/d0bd01beca0308cfb263ec7f44c9acf0933a1fc0/src/_pytest/python.py#L204
If you remove that, then you have to explicitly add the `__init__.py` pattern to "python_files" to enable the Package collection. I'm not sure what's the best way to look for packages but also not duplicate the filename matching.

Reply:
Gentle ping here? I'm not sure if it is worth reusing the filename matching for `__init__` files, which is what I got from @RonnyPfannschmidt's comment.

Reply:
packages have different behaviours than modules to begin with, so i dont see a problem there

Reply:
So we can go ahead and use my suggestion?

```python
init_module = this_path.join("__init__.py")
if init_module.check(file=1) and init_module.match(self.config.getini('python_files')):
    yield Module(init_module, self)
```

Reply:
yes